packages/firestore/src/remote/serializer.ts

```diff
@@ -99,15 +99,13 @@ const OPERATORS = (() => {
 })();
 
 function assertPresent(value: unknown, description: string): asserts value {
-  assert(!typeUtils.isNullOrUndefined(value), description + ' is missing');
+  assert(!isNullOrUndefined(value), description + ' is missing');
 }
 
-// This is a supplement to the generated proto interfaces, which fail to account
-// for the fact that a timestamp may be encoded as either a string OR this.
-interface TimestampProto {
-  seconds?: string | number;
-  nanos?: number;
-}
+// Denotes the possible representations for timestamps in the Value type.
+export type TimestampValue =
```

Comment:
The `Value` proto can contain a timestamp and the name `TimestampValue` suggests that this is one of those, when it's really just the timestamp itself. Also, this is somewhat confusing because historically FooValue has been one of our model types and not a proto. In this regard the old `TimestampProto` name was better.

Reply:
I renamed it to Timestamp, which aligns with LatLng and GeoPoint. I also moved it to the proto d.ts file.